File: 3931-check-adjacent-digit-differences/3931-check-adjacent-digit-differences.c
bool isAdjacentDiffAtMostTwo(char* s) {
    for (int i = 1; s[i] != '\0'; i++) {
        int diff = s[i] - s[i - 1];
        if (diff < 0) {
            diff = -diff;
        }
        if (diff > 2) {
            return false;
        }
    }
    return true;
}

File: 3931-check-adjacent-digit-differences/3931-check-adjacent-digit-differences.rs
impl Solution {
    pub fn is_adjacent_diff_at_most_two(s: String) -> bool {
        let bytes = s.as_bytes();
        for i in 1..bytes.len() {
            let diff = (bytes[i] as i32 - bytes[i - 1] as i32).abs();
            if diff > 2 {
                return false;
            }
        }
        true
    }
}